lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2206,6 +2206,8 @@ void generateIntelFPGAAnnotation(const SPIRVEntry *E,
     Out << "{pump:1}";
   if (E->hasDecorate(DecorationDoublepumpINTEL))
     Out << "{pump:2}";
+  if (E->hasDecorate(DecorationUserSemantic))
```

Comment:
This decoration is a core spec feature and it is not related to intelfpga annotations. So, it looks a little bit strange to handle core functionality in this function

Reply:
Agree. But now I have no ideas how to add this feature and not to duplicate whole `transIntelFPGADecorations` function, which functionality is fully used for UserSemantic translation to LLVM-IR.
Thinking about removing `IntelFPGA` from related function names, but it's cool solution too.